server/src/main/java/io/druid/server/coordinator/CostBalancerStrategy.java

```diff
@@ -19,44 +19,36 @@
 
 package io.druid.server.coordinator;
 
+import com.google.common.base.Predicates;
+import com.google.common.collect.Iterables;
 import com.google.common.collect.Lists;
 import com.google.common.util.concurrent.Futures;
 import com.google.common.util.concurrent.ListenableFuture;
 import com.google.common.util.concurrent.ListeningExecutorService;
-import com.google.common.util.concurrent.MoreExecutors;
 import com.metamx.common.Pair;
 import com.metamx.emitter.EmittingLogger;
 import io.druid.timeline.DataSegment;
-import org.joda.time.DateTime;
+import org.apache.commons.math3.util.FastMath;
 import org.joda.time.Interval;
 
 import java.util.List;
 import java.util.concurrent.Callable;
-import java.util.concurrent.ExecutorService;
 
 public class CostBalancerStrategy implements BalancerStrategy
 {
   private static final EmittingLogger log = new EmittingLogger(CostBalancerStrategy.class);
-  private static final long DAY_IN_MILLIS = 1000 * 60 * 60 * 24;
-  private static final long SEVEN_DAYS_IN_MILLIS = 7 * DAY_IN_MILLIS;
-  private static final long THIRTY_DAYS_IN_MILLIS = 30 * DAY_IN_MILLIS;
-  private final long referenceTimestamp;
-  private final ListeningExecutorService exec;
 
-  public static long gapMillis(Interval interval1, Interval interval2)
-  {
-    if (interval1.getStartMillis() > interval2.getEndMillis()) {
-      return interval1.getStartMillis() - interval2.getEndMillis();
-    } else if (interval2.getStartMillis() > interval1.getEndMillis()) {
-      return interval2.getStartMillis() - interval1.getEndMillis();
-    } else {
-      return 0;
-    }
-  }
+  static final double HALF_LIFE = 24.0; // cost function half-life in hours
```

Comment:
it'd be really nice to have some comments about what everything means and how the algo works